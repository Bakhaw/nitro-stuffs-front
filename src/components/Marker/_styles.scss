.Marker {
  cursor: pointer;
  &:hover {
    transform: scale(1.1);
    transition: 0.2s;
  }
}
