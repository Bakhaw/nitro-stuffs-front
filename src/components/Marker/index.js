import React from 'react';
import PropTypes from 'prop-types';

import Image from '../Image';

function Marker({ additionalStyles, x, y, height, width, onClick, type }) {
  return (
    <div
      className='Marker'
      style={{
        position: 'absolute',
        left: x,
        top: y,
        ...additionalStyles,
      }}
      onClick={onClick}
    >
      <Image width={width} height={height} iconType={type} />
    </div>
  );
}

Marker.propTypes = {
  additionalStyles: PropTypes.object,
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  height: PropTypes.string,
  width: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired,
};

export default Marker;
