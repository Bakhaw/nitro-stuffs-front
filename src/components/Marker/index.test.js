import React from 'react';
import { shallow } from 'enzyme';

import Marker from '.';

test('should render properly', () => {
  const props = {
    x: 15,
    y: 250,
    height: 50,
    width: 50,
    onClick: jest.fn(),
    type: 'position',
  };

  const container = shallow(<Marker {...props} />);
  expect(container.debug()).toMatchSnapshot();
});
