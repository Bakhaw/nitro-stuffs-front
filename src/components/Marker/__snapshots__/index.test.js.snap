// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
"<div className=\\"Marker\\" style={{...}} onClick={[Function: mockConstructor]}>
  <Image width={50} height={50} iconType=\\"position\\" />
</div>"
`;
